src/provider/metamask/MetaMask.ts

```diff
@@ -329,7 +329,7 @@ export class MetaMask {
         gas: authInfo.fee?.gasLimit.toString() ?? "0",
       }
       const aminoMsgs = msgs.map(msg => AminoTypesMap.toAmino(msg))
-      const sig = await metamask.signEip712(
+      const { sig } = await metamask.signEip712(
```

Comment:
Updated it to use new doc, before was because the type did not match, the doc required in signDirect was Models.Tx.SignDoc but the doc EIP712 gave was CarbonTx.StdSignDoc

but i have converteed the stdsigndoc into models.tx.signdoc now

Reply:
oh doesnt matter mah, cos what we need is the signature only ie a string for signdirect anyway

Reply:
we need the memo for the signdoc to be passed through also though that's why need to add the new doc

the memo will include the carbon chain id to prevent replay attack

Reply:
hmm then in that case why for signDirect the doc is not regenerated ah, since we need to append the chain id and regenerate a new memo?

Reply:
Ah yeah should be regenerated, just added it but metamask is also deprecated right?

Reply:
Yes